### ISSUE TITLE
Using `$`

### ISSUE THREAD

**Issue by mikearnaldi** [Effect-TS/io#403]
We may want to stop using `$` in generators as it has special meaning under https://qwik.builder.io/docs/advanced/dollar/ 

The risk is (and I am not even sure it is true) that a consumer of effect using qwik will end up with potentially unknown behaviour

**Comment by mikearnaldi** [comment]
> `$` is used in many frameworks and libraries, like jQuery, facebook/lexical, etc. It's up to the Qwik not to break existing code.

same as always, whatever we can do to prevent breaking user code should be done at the lib level, I don't care if the user uses `$` in their code.

**Comment by steida** [comment]
Then I would use `_`. It even looks better.

**Comment by mikearnaldi** [comment]
I was using `_` before, personally I like it more as it focused the attention better on the argument `yield* _(hello)` , also in lib code we hardly ever use `gen` I don't think it's ever used in `/io` for example so this is kind of a preventive note rather than a refactor request

**Comment by patroza** [comment]
`_` is overloaded already because of mimicking Scala's shorthand lambda syntax.. 
I haven't tested qwik, but `$` is technically not something suffixed by `$`..

**Comment by IMax153** [comment]
@mikearnaldi - we do use `Effect.gen` for tests, which a lot of folks are currently looking at for examples and such. 

I personally don't see any issue with using `$`. But whatever we decide we should document in a section of the documentation about writing idiomatic Effect code. 

cc @gcanti @schickling

**Comment by patroza** [comment]
> `$` is used in many frameworks and libraries, like jQuery, facebook/lexical, etc. It's up to the Qwik not to break existing code.

I agree, this has been used for decades by many. for some library like qwik now to hijack it with specialised meaning...
What do we do on the next library that comes along and hijacks something else?

**Comment by tim-smart** [comment]
I think how `$` and `_` are used in the ecosystem is a non-argument. They are both used in the wild by different libraries and frameworks for different reasons.

I have been using `_` recently to give it some milage and test whether it is has any benefit over `$`, and I have found that it makes reading `Effect.gen` slightly less noisy.

I personally think we should make the switch to `_`, anything to make `Effect.gen` more approachable is worth it IMO.

**Comment by IMax153** [comment]
> I personally think we should make the switch to `_`, anything to make `Effect.gen` more approachable is worth it IMO.

I agree with this, but only for documentation. I think swapping from `$` to `_` in the tests and whatnot in the codebase would be bikeshedding at this point.

**Comment by aniravi24** [comment]
I typically use underscores to represent unused variables and function arguments, and also use eslint/TS compiler options to enforce that.

**Comment by patroza** [comment]
@aniravi24 yea that is everywhere the standard, except in Scala or Effect :D

**Comment by IMax153** [comment]
Closing this issue as I think that between the discussion in Discord and what was discussed here, we generally agree that there is no reason to restrict `$`.
